### PR TITLE
Fix test running

### PR DESCRIPTION
**Pull request by psychon**
The old hack with TESTS_ENVIRONMENT no longer works. I do not know which
automake version it worked with, but it causes problems now. So, this
commit instead uses a LOG_COMPILER for the same job.

Signed-off-by: Uli Schlachter <psychon@znc.in>

----

@osch Is this okay with you?

I also briefly tried making the test suite work with Lua 5.3, but... meh, quickly gave up. It might be a good idea to set up Travis CI integration for this, but... also meh.

### CODE REVIEW

File: Makefile.am [hunk 1]
```diff
@@ -32,7 +32,7 @@ TESTS += test/surface.lua
 TESTS += test/svg_surface.lua
 TESTS += test/region.lua
 EXTRA_DIST += examples/images/pattern.png
-EXTRA_DIST += $(TESTS) lunit.lua test-setup.lua lunit-console.lua test-loading.lua
+EXTRA_DIST += $(TESTS) lunit.lua test-setup.lua lunit-console.lua test-loading.lua run-tests.sh
```
Comment (osch) [inline Makefile.am]: shouldn't this be `run-test.sh` instead of `run-tests.sh`? Or the file below should be renamed?

Comment (osch) [reply Makefile.am]: I'm not an gnu autotools expert (although I used autotools in the past)... Ah now I found it: the tests can be started via `make check` not via `make test` as it is written in the readme. After `make check` with Lua 5.3 the result is:

```============================================================================
Testsuite summary for oocairo 1.5
============================================================================
# TOTAL: 14
# PASS:  14
# SKIP:  0
# XFAIL: 0
# FAIL:  0
# XPASS: 0
# ERROR: 0
============================================================================
```

So it seems to me that the testsuite now does run ok with Lua 5.3.

Comment (psychon) [reply Makefile.am]: > I'm not an gnu autotools expert (although I used autotools in the past)..

Heh. Stackoverflow told me how to get this to work. :-)

I added a new commit updating the README. Thanks for mentioning this problem. And also thanks for testing with Lua 5.3. I guess Debian disables some deprecated stuff in Lua 5.3 (the problem is that `module()` is missing).

**Comment by osch** [comment]
> testing with Lua 5.3. I guess Debian disables some deprecated stuff in Lua 5.3 (the problem is that module() is missing)

I checked this: under arch/manjaro lua is compiled with `-DLUA_COMPAT_5_2 -DLUA_COMPAT_5_1`. With this the `module` function is available. With the plain lua Makefile theses defines are not set so `module` is not available under plain Lua 5.3.

**Pull request merged**